File: 6.z-字形变换.cpp
/*
 * @lc app=leetcode.cn id=6 lang=cpp
 *
 * [6] Z 字形变换
 */

// @lc code=start
class Solution {
public:
    string convert(string s, int numRows) {
        if(numRows ==1 || numRows >=s.size()||s.size()==1){
            return s;
        }     
        vector<string> rows(numRows);
        int flag=1,index=0;
        for(int i=0;i<s.size();i++){
            rows[index]+=s[i];
            if(index==numRows-1){
                flag=-1;
            }else if(index==0){
                flag=1;
            }
            index+=flag;
        }
        return accumulate(rows.begin(),rows.end(),string(""));
    }
};
// @lc code=end

File: 6.z-字形变换.py
#
# @lc app=leetcode.cn id=6 lang=python3
#
# [6] Z 字形变换
#

# @lc code=start
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1 or numRows >= len(s):
            return s
        rows=['']*numRows
        flag,cur_row=0,0
        for c in s:
            rows[cur_row]+=c
            if cur_row==0:
                flag=1
            elif cur_row==numRows-1:
                flag=-1
            cur_row+=flag
        return ''.join(rows)

# @lc code=end

